docs/metrics-overview.md

```diff
@@ -29,6 +29,116 @@ The COSI driver exposes default gRPC server metrics to monitor RPC activity.
+
+### IAM Operations
+
+| IAM Operation          | Description                                                         |
+|-------------------------|---------------------------------------------------------------------|
+| `CreateUser`           | Creates an IAM user with the specified username.                   |
+| `CreateAccessKey`      | Generates access keys for a specific IAM user.                     |
+| `PutUserPolicy`        | Attaches an inline S3 wildcard policy to a user for bucket access. |
+| `GetUser`              | Retrieves details about an IAM user.                               |
+| `ListAccessKeys`       | Lists all access keys associated with an IAM user.                 |
+| `DeleteAccessKey`      | Deletes a specific access key associated with an IAM user.         |
+| `DeleteUserPolicy`     | Deletes an inline policy associated with an IAM user.              |
+| `DeleteUser`           | Deletes an IAM user.                                               |
+
+### Example IAM Metrics Output
+
+Duration of IAM requests in seconds
+
+```sh
+scality_cosi_driver_iam_request_duration_seconds_bucket{method=“CreateUser”,status=“success”,le=“0.01”} 3
+scality_cosi_driver_iam_request_duration_seconds_bucket{method=“CreateUser”,status=“success”,le=“0.025”} 4
+scality_cosi_driver_iam_request_duration_seconds_sum{method=“CreateUser”,status=“success”} 0.014
+scality_cosi_driver_iam_request_duration_seconds_count{method=“CreateUser”,status=“success”} 4
+```
+
+Total number of IAM requests
+
+```sh
+scality_cosi_driver_iam_requests_total{method=“CreateUser”,status=“success”} 4
+scality_cosi_driver_iam_requests_total{method=“DeleteAccessKey”,status=“error”} 1
+```
+
+### Example IAM Workflow
+
+#### Creating Bucket Access
+
+1. Create an IAM user (`CreateUser`).
+2. Attach an inline policy for bucket access (`PutUserPolicy`).
+3. Generate access keys for the IAM user (`CreateAccessKey`).
+
+#### Revoking Bucket Access
+
+1. Verify the IAM user exists (`GetUser`).
+2. Delete inline policies (`DeleteUserPolicy`).
+3. Delete all associated access keys (`DeleteAccessKey`).
+4. Delete the IAM user (`DeleteUser`).
+
+---
+
+## S3 Operation Metrics
+
+The COSI driver collects metrics for S3 bucket operations performed via the AWS S3 API. These metrics help monitor bucket-related operations and their durations.
+
+### Key S3 Metrics
+
+| Metric Name                                        | Description                                                | Labels            | Example Values                  |
+|---------------------------------------------------|------------------------------------------------------------|-------------------|----------------------------------|
+| `scality_cosi_driver_s3_request_duration_seconds` | Histogram of S3 request durations in seconds.             | `method`, `status`| `CreateBucket`, `success`, `error`|
```

Comment:
ditto about mentioning only `success` or `error`

Reply:
Changed in https://github.com/scality/cosi-driver/pull/85/commits/989ef11ce706ab73b764fb4e338846eec98be67e